src/uct/ib/rdmacm/rdmacm_md.c

```diff
@@ -49,16 +49,18 @@ ucs_status_t uct_rdmacm_md_query(uct_md_h md, uct_md_attr_t *md_attr)
     return UCS_OK;
 }
 
-static int uct_rdmacm_get_event_type(struct rdma_event_channel *event_ch)
+static enum rdma_cm_event_type uct_rdmacm_get_event_type(struct rdma_event_channel *event_ch)
 {
     struct rdma_cm_event *event;
-    int ret, event_type;
+    int ret;
+    enum rdma_cm_event_type event_type;
 
     /* Fetch an event */
     ret = rdma_get_cm_event(event_ch, &event);
     if (ret) {
         ucs_warn("rdma_get_cm_event() failed: %m");
-        return 0;
+        UCS_STATIC_ASSERT(RDMA_CM_EVENT_ADDR_RESOLVED == 0);
```

Comment:
why need this static assert?

Reply:
original code returned 0 as error value which caused compiler error. I replaced it by enum value & added assert to be sure that original logic was not changed.

removed assert